apps/dashboard/src/components/billing/plan-action-button.tsx

```diff
@@ -0,0 +1,94 @@
+import { Button } from '@/components/primitives/button';
+import { ApiServiceLevelEnum } from '@novu/shared';
+import { useMutation } from '@tanstack/react-query';
+import { get, post } from '../../api/api.client';
+import { toast } from 'sonner';
+import { useSubscription } from './hooks/use-subscription';
+import { cn } from '../../utils/ui';
+import { useTelemetry } from '../../hooks/use-telemetry';
+import { TelemetryEvent } from '../../utils/telemetry';
+
+interface PlanActionButtonProps {
+  selectedBillingInterval: 'month' | 'year';
+  variant?: 'default' | 'outline';
+  showIcon?: boolean;
+  className?: string;
+  size?: 'default' | 'sm' | 'lg';
+}
+
+export function PlanActionButton({
+  selectedBillingInterval,
+  variant = 'default',
+  className,
+  size = 'default',
+}: PlanActionButtonProps) {
+  const { data, isLoading } = useSubscription();
+  const track = useTelemetry();
+
+  const isPaidSubscriptionActive = () => {
+    return data?.isActive && !data?.trial?.isActive && data?.apiServiceLevel !== ApiServiceLevelEnum.FREE;
+  };
+
+  const { mutateAsync: checkout, isLoading: isCheckingOut } = useMutation(
```

Comment:
Nit-picky one, consider introducing a use-checkout hook so that this is reusuable.